### PR TITLE
folder structure/webpack

### PR DESCRIPTION
**Pull request by Willian1661**
@eerison  may you see what can i change about the folders, please

### CODE REVIEW

**Comment by Willian1661** [comment]
i have already saw one thing 🤣

**Comment by eerison** [comment]
it`s wrong

- why do you create the folder app?
- you dont need the jest folder, where did you see this? and package files must be in the root!
- your css must be into the assets and not layouts! layout folder is used for keep more then one layout
- You write calc in portuguese, and it must be into page folder
- Where is your index file? do you know that the main page is named as `index`?

FYI: if the pipeline fails it will be never merged :)

Edit 1: you can check [this article](https://dev.to/vadorequest/a-2021-guide-about-structuring-your-next-js-project-in-a-flexible-and-efficient-way-472)

Edit 2: font 2: 
https://blog.dennisokeeffe.com/blog/2021-12-06-nextjs-enterprise-project-structure
@Willian1661

**Comment by Willian1661** [comment]
i commited also this **.github** that has the workflow's file
i needed commited as well or nevermind ??

**Comment by eerison** [comment]
> i commited also this **.github** that has the workflow's file i needed commited as well or nevermind ??

? I dont get it! I dont see any changes into the .github folder!

**Comment by Willian1661** [comment]
"" Edit 1: you can check [this article](https://dev.to/vadorequest/a-2021-guide-about-structuring-your-next-js-project-in-a-flexible-and-efficient-way-472)

Edit 2: font 2:
https://blog.dennisokeeffe.com/blog/2021-12-06-nextjs-enterprise-project-structure ""
just follow the structure or also create a next-app ?

**Comment by eerison** [comment]
> "" Edit 1: you can check [this article](https://dev.to/vadorequest/a-2021-guide-about-structuring-your-next-js-project-in-a-flexible-and-efficient-way-472)
> 
> Edit 2: font 2: https://blog.dennisokeeffe.com/blog/2021-12-06-nextjs-enterprise-project-structure "" just follow the structure or also create a next-app ?

just follow the structure as we talked 😵‍💫

**Comment by Willian1661** [comment]
i created the jest folder because here:
Screenshot from 2022-12-15 09-41-13
under the utils has a test folder, so i created the "jest" folder to be the test one or its better to be just "test"?

**Comment by eerison** [comment]
> i created the jest folder because here: Screenshot from 2022-12-15 09-41-13 under the utils has a test folder, so i created the "jest" folder to be the test one or its better to be just "test"?

test is better

**Comment by eerison** [comment]
assets and pages should be inside of src

rename you image to background

**Comment by eerison** [comment]
use app.css and app.js

or main.css and main.js and let it inside of assets

**Comment by eerison** [comment]
> it`s wrong
> 
> * why do you create the folder app?
> * you dont need the jest folder, where did you see this? and package files must be in the root!
> * your css must be into the assets and not layouts! layout folder is used for keep more then one layout
> * You write calc in portuguese, and it must be into page folder
> * Where is your index file? do you know that the main page is named as `index`?
> 
> FYI: if the pipeline fails it will be never merged :)
> 
> Edit 1: you can check [this article](https://dev.to/vadorequest/a-2021-guide-about-structuring-your-next-js-project-in-a-flexible-and-efficient-way-472)
> 
> Edit 2: font 2: https://blog.dennisokeeffe.com/blog/2021-12-06-nextjs-enterprise-project-structure @Willian1661

public is in root no into src

**Comment by eerison** [comment]
assets and public folder is a bit confuse , wdyt to use assets only

**Comment by Willian1661** [comment]
> assets and public folder is a bit confuse , wdyt to use assets only

my public is into the root and my assets into src

**Comment by eerison** [comment]
> > assets and public folder is a bit confuse , wdyt to use assets only
> 
> my public is into the root and my assets into src

i know, but i didnt ask it

**Comment by eerison** [comment]
main.js into assets
index.html into pages

@Willian1661

**Comment by Willian1661** [comment]
> main.js into assets index.html into pages

calm down
i have a src, assets, pages folder, ok.

- index.html is **src/index.html** ; main.js is **pages/main.js** and main.css is **assets/main.css**, ok

but instead of this, it's to be :

- **src/pages/index.html**;  **src/assets/main.js**;  **src/assets/main.css**;

right? that's it?

@eerison

**Comment by eerison** [comment]
> src/pages/index.html; src/assets/main.js; src/assets/main.css;

yep

**Comment by Willian1661** [comment]
> > src/pages/index.html; src/assets/main.js; src/assets/main.css;
> 
> yep

only this?

**Comment by eerison** [comment]
it looks correct for me, now its missing you setup webpack :D

**Comment by Willian1661** [comment]
> it looks correct for me, now its missing you setup webpack :D

let's for webpack  😝

**Comment by Willian1661** [comment]
@eerison 
for the webpack file i put it into the root or into assets folder, have any difference ?

**Comment by eerison** [comment]
> @eerison for the webpack file i put it into the root or into assets folder, have any difference ?

everyconfig file usually is into the root

**Comment by Willian1661** [comment]
@eerison 
how i know webpack it's working?

**Comment by eerison** [comment]
When you build using webpack you should see the files generated into dist folder

**Comment by eerison** [comment]
@Willian1661 

Other thing, usually I see the test in the same folder as the js file 

Like this: https://twitter.com/Mokkapps/status/1603433508478849024?t=Xj3ULCTjLzfk0AC9i0l3rg&s=19

I guess you should move sum.test.js to src,  and keep booth files together!

**Comment by Willian1661** [comment]
@eerison 
i finished webpack setup, but it's some things that is wrong
beucause i can't acess the main.js that it creates that should be this:
Screenshot from 2022-12-16 19-16-41

and i still can see my Main.js file

**Comment by Willian1661** [comment]
i created a test folder that works(i believe the way it should):
Screenshot from 2022-12-16 20-24-10
**inside this src folder it's the index.js file**

i need to follow this exactly way?
i created like this:
Screenshot from 2022-12-16 20-29-32

**Comment by eerison** [comment]
Dist folder is generated automatically,  we don't commit folders like this! 

It's like node_modules

**Comment by eerison** [comment]
Are you checking the index generated into the dist?

**Comment by eerison** [comment]
You must use the same filename partner 

You are using 
Main.js
main.css

**Comment by Willian1661** [comment]
> You must use the same filename partner

no i don't think so, 'cause i installed the dependencies  for webpack and not createed the **dist** folder.
i'am reading the webpack  **getting started** and it says to create the dist folder by hand: https://webpack.js.org/guides/getting-started/#creating-a-bundle

**Comment by Willian1661** [comment]
> You must use the same filename partner
> 
> You are using Main.js main.css

i forgot to change this one 😅

**Comment by Willian1661** [comment]
> Dist folder is generated automatically, we don't commit folders like this!
> 
> It's like node_modules

but i don't understood one thing, inside the dist file it has a index.html,ok?
so this file it's suppose to be my html.file? the html of the page?

**Comment by Willian1661** [comment]
> Are you checking the index generated into the dist?

into the dist it's not generated an index file, i have to create
so i thought "if a need to create one, i think it's better to use the one i already have"

**Comment by Willian1661** [comment]
> Are you checking the index generated into the dist?

into the dist folder it's just generated a main.js, the one a seted on the output on webpack.config.js

**Comment by Willian1661** [comment]
> @eerison i finished webpack setup, but it's some things that is wrong beucause i can't acess the main.js that it creates that should be this: Screenshot from 2022-12-16 19-16-41
> 
> and i still can see my Main.js file

what did different now:
1.moved the main.js generated by webpack into pages, because the index.html is into pages too. 
2. and know it's working the main.js from webpack

**Comment by eerison** [comment]
again, you shouldn't commit dist folder!

**Comment by Willian1661** [comment]
> again, you shouldn't commit dist folder!

i know, i didn't now
it's inside .gitignore now

**Comment by eerison** [comment]
> > again, you shouldn't commit dist folder!
> 
> i know, i didn't now it's inside .gitignore now

It's still there,  anyway when it's avaliable to review ,just request it ;)

**Comment by Willian1661** [comment]
> > > again, you shouldn't commit dist folder!
> > 
> > 
> > i know, i didn't now it's inside .gitignore now
> 
> It's still there, anyway when it's avaliable to review ,just request it ;)

it's still there because i didn't commit yet
i'll request next time

File: src/assets/index.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+import export_test from "./Main";
```
Comment (Willian1661) [inline src/assets/index.js]: @eerison 
it's ok to use this single file just for this?

File: src/assets/Main.js [hunk 2]
```diff
@@ -65,3 +65,8 @@ function gerenciador() {
     });
 }
 gerenciador();
+
```
Comment (eerison) [inline src/assets/Main.js]: every single file is lowcase and you choosed to keep those 2 file CamelCase!

change to lowcase as well!

File: src/assets/index.js (hunk 1 above)
Comment (eerison) [reply src/assets/index.js]: why do you need this index.js? IMO it's unnecessary

File: src/pages/index.html [hunk 3]
```diff
@@ -6,11 +6,11 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <title>Calculator </title>
-    <link rel="stylesheet" href="css11.css">
+    <link rel="stylesheet" href="/src/assets/Main.css">
```
Comment (eerison) [inline src/pages/index.html]: it should be from dist folder, shouldn't it?

File: webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+const path = require('path');
+
+module.exports = {
+  mode: 'development',
+    entry:{
+      bundle: './src/assets/index.js',
```
Comment (eerison) [inline webpack.config.js]: hmmm I get it! But I guess you can set more then one file as entry, can'r you?

File: src/pages/index.html (hunk 3 above)
Comment (Willian1661) [reply src/pages/index.html]: > it should be from dist folder, shouldn't it?

aah that's true, the dist it's suppose to have all the assets file?
like js and css

File: webpack.config.js (hunk 4 above)
Comment (Willian1661) [reply webpack.config.js]: yea it's works like: 
entry:{
      bundle: './src/assets/index.js',
        main: './src/assets/index.js',
          otherName: './src/assets/index.js',
}

Comment (eerison) [reply webpack.config.js]: then you don't need the index.js!

File: src/pages/index.html (hunk 3 above)
Comment (eerison) [reply src/pages/index.html]: yep

File: webpack.config.js (hunk 4 above)
Comment (Willian1661) [reply webpack.config.js]: why not? it says that the entry file(the index.js) will be named bundle by webpack.config
and the index.js it's the file i'am doing the export to the Main.js

Comment (Willian1661) [reply webpack.config.js]: if you're saying i don't need it.
the path the file have to do it's for the Main.js ?🤔
**like:**

entry:{
bundle: './src/assets/**Main.js**',
}

Comment (eerison) [reply webpack.config.js]: yep it'll be renamed, then if you use xyz.js it will be build using index.js

**Comment by Willian1661** [comment]
@eerison 
it's wrong my public folder be inside of src?

**Comment by Willian1661** [comment]
@eerison 
take a look, i think it's all done(for now ofc) :)

**Comment by Willian1661** [comment]
@eerison 
while i was configuring the webpack i was doing all in another folder "test" that i created
and when was all good in that folder, finally i passed everything to the original folder.
that's a good way to do it things like this? i did this because i was fear to loose any file or important stuffs.

File: webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    },
+    output: {
+      path: path.resolve(__dirname, 'dist'),
+      filename: '[name].js',
+      clean: true,
+  },
+    plugins: [
+      new HtmlWebpackPlugin({
+        filename: 'bundle.html',
+        template: './src/pages/index.html',
```
Comment (eerison) [inline webpack.config.js:17]: Rename to index.html, because it's your index file;)

File: webpack.config.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  module: {
+    rules: [
+      {
+        test: /\.css$/i,
+        use: ['style-loader', 'css-loader'],
+      },
+      {
+        test: /\.(png|svg|jpg|jpeg|gif)$/i,
+        type: 'asset/resource',
+      },
```
Comment (eerison) [inline webpack.config.js:29]: I guess it should be assets/images,  don't you?

Comment (Willian1661) [reply webpack.config.js:29]: but it's the type ,not the path

Comment (eerison) [reply webpack.config.js:29]: Hmmm

File: webpack.config.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+      bundle: './src/assets/main.js',
+    },
+    output: {
+      path: path.resolve(__dirname, 'dist'),
+      filename: '[name].js',
+      clean: true,
+  },
+    plugins: [
+      new HtmlWebpackPlugin({
+        filename: 'bundle.html',
```
Comment (eerison) [inline webpack.config.js]: index.html

Comment (eerison) [inline webpack.config.js]:
```suggestion
        filename: 'index.html',
```

**Pull request merged**